### ISSUE TITLE
Missing "link" between field or group and alert

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#5932]
Currently, we never link the alert (`<xf:alert>` in the source, `.xforms-alert` in the HTML) with the field or group. This could be done by always putting an `id` on the alert, and adding this id to the `aria-errormessage`, in a way similar to what we do for the hint.

Can we set the `aria-errormessage` ([MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage)) when the HTML is initially produced, and then keep it unchanged when the validity of the control changes? The message in the element pointed to by `aria-errormessage` might be read even if that element is hidden. This might be ok in the case of Form Runner if the element is always empty when the alert message isn't shown, but it might be a problem for forms where the XForms is written by hand, which we might consider being beyond the scope of this issue.

Related issue: #5929

[+1 from customer](https://3.basecamp.com/3600924/buckets/1966718/messages/6338049163)

**Comment by ebruchez** [comment]
Since we are able to do this for `aria-labelledby` and `aria-describedby`, it shouldn't be too hard to do this.

**Comment by ebruchez** [comment]
- [ ] cherry-pick on 2022.1-pe
